extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/linkHandler/PeertubeChannelLinkHandlerFactory.java

```diff
@@ -31,11 +31,17 @@ public String getUrl(String id, List<String> contentFilters, String searchFilter
     @Override
     public String getUrl(String id, List<String> contentFilter, String sortFilter, String baseUrl)
             throws ParsingException {
-        return baseUrl + ACCOUNTS_ENDPOINT + id;
+
+        if (id.matches(ID_PATTERN)) {
+            return baseUrl + API_ENDPOINT + id;
+        } else {
+            // This is needed for compatibility with older versions were we didn't support video channels yet
+            return baseUrl + API_ENDPOINT + "accounts/" + id;
```

Comment:
I don't understand why this would be needed. Could you provide a url where this piece of code would run?

Reply:
@Royosef

Reply:
It's for compatibility with older versions of NewPipeExtractor, where you had just e.g. `framasoft` as ID. Now that ID would be `accounts/framasoft`. If you had stored an old ID such as `framasoft` and try to create an extractor from the old ID, it would fail. This code prefixes those old IDs with `accounts/` in the URL, so old IDs still work. NewPipe doesn't store IDs but URLs though.